packages/pyright-internal/src/analyzer/operations.ts

```diff
@@ -215,34 +220,46 @@ export function validateBinaryOperation(
             return { type: NeverType.createNever() };
         }
 
+        const validateArithmetic = (context: InferenceContext | undefined, addendum: DiagnosticAddendum | undefined) =>
+            validateArithmeticOperation(
+                evaluator,
+                operator,
+                leftTypeResult,
+                rightTypeResult,
+                errorNode,
+                context,
+                addendum,
+                options
+            );
+
         // Handle certain operations on certain homogenous literal types
         // using special-case math. For example, Literal[1, 2] + Literal[3, 4]
         // should result in Literal[4, 5, 6].
         if (options.isLiteralMathAllowed) {
             type = calcLiteralForBinaryOp(operator, leftType, rightType);
+            const result = validateArithmetic(undefined, undefined);
+            if (type) {
+                overloadsUsedForCall.push(...(result?.overloadsUsedForCall ?? []));
```

Comment:
how come we get 2 types here? cant we just get the type from the existing `calcLiteralForBinaryOp` and use that instead?
```suggestion
            type = calcLiteralForBinaryOp(operator, leftType, rightType);
            if (type) {
                overloadsUsedForCall.push(...type.overloadsUsedForCall);
```

Reply:
The appreciation is much appreciated! I found basedpyright by chance after using my own Pyright fork with semantic tokens for a while and I am especially thankful that you maintain this great fork (and that I don’t have to keep my old fork up-to-date anymore 😉)

Your extended version of `calcLiteralForBinaryOp` is certainly more efficient than the previous implementation, thanks!